azurelinuxagent/common/cgroups.py

```diff
@@ -404,7 +405,9 @@ def __init__(self, name, path_maker):
             self.is_wrapper_cgroup = False
 
         self.cgroups = {}
-        self.threshold = None
+
+        self.threshold = CGroupsLimits(self.name, cpu_threshold=limits["cpu"],
+                                       memory_threshold=limits["memory"]) if limits else CGroupsLimits(self.name)
```

Comment:
This will throw an exception if limits["cpu"] or limits["memory"] isn't defined. The problem is the design of the CGroupsLimits constructor; see my comment there.